test/functional/page_objects/time_picker.ts

```diff
@@ -125,7 +125,28 @@ export class TimePickerPageObject extends FtrService {
    * @param {String} toTime MMM D, YYYY @ HH:mm:ss.SSS
    */
   public async setAbsoluteRange(fromTime: string, toTime: string) {
-    this.log.debug(`Setting absolute range to ${fromTime} to ${toTime}`);
+    // setAbsoluteRange used to always set the timepicker.  This version checks if the
+    // required times are already in the URL and, if so, returns to save testing time.
+    const currentUrl = await this.browser.getCurrentUrl();
+    const DEFAULT_DATE_FORMAT = 'MMM D, YYYY @ HH:mm:ss.SSS';
+    const startMoment = moment.utc(fromTime, DEFAULT_DATE_FORMAT).toISOString();
+    const endMoment = moment.utc(toTime, DEFAULT_DATE_FORMAT).toISOString();
+    if (currentUrl.includes(startMoment) && currentUrl.includes(endMoment)) {
```

Comment:
Maybe I'm late to jump here, but just one thought. Maybe we can make a more strict times check?
```
currentUrl.includes(`time:(from:'${startMoment}',to:'${endMoment}')`)
```
I guess the format is consistent across the plugins, right?

Reply:
Before I even read the rest of this, I wonder if it's worth to strongly type the format.  
Actually, I'm hoping that's been done somewhere in this repo already.

Reply:
I got @dmlemeshko 's suggestion to work.  I had to decodeURI.